geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+    // point and multipoint disabled when freehand tool selected
+    pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+    multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+
+    // only show the selected geometry type as enabled while editing
+    geometryEditor.startedProperty().addListener((ob, ov, nv) -> {
+      if (geometryEditor.isStarted()) {
+        pointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POINT)));
+        multipointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.MULTIPOINT)));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYLINE)));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYGON)));
+      } else {
+        // when editor stopped, re-enable polygon & polyline, enable point/multipoint if vertex tool selected
+        pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(false));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(false));
+      }
+    });
+
+    // bind the geometry editor tool to the tool chosen in the combo box
+    toolComboBox.valueProperty().bindBidirectional(geometryEditor.toolProperty());
+
+    // undo/redo enabled when the geometry editor can undo/redo
+    undoButton.disableProperty().bind(geometryEditor.canUndoProperty().not());
+    redoButton.disableProperty().bind(geometryEditor.canRedoProperty().not());
+
+    // delete selected element enabled only when an element is selected and deletable
+    ChangeListener<GeometryEditorElement> selectedElementChangedListener = (ob, ov, nv) -> {
+      if (nv == null) {
+        deleteSelectedElementButton.disableProperty().bind(new SimpleBooleanProperty(true));
+      } else {
+        deleteSelectedElementButton.disableProperty().bind(geometryEditor.getSelectedElement().canDeleteProperty().not());
+      }
+    };
+    geometryEditor.selectedElementProperty().addListener(selectedElementChangedListener);
+
+    // save/discard buttons enabled when geometry editor has started
+    stopAndSaveButton.disableProperty().bind(geometryEditor.startedProperty().not());
+    stopAndDiscardButton.disableProperty().bind(geometryEditor.startedProperty().not());
+
+    // delete all geometries button enabled when geometry editor stopped
+    deleteAllGeometriesButton.disableProperty().bind(geometryEditor.startedProperty());
```

Comment:
```suggestion
    deleteAllGeometriesButton.disableProperty().bind(
        Bindings.or(geometryEditor.startedProperty(), Bindings.createBooleanBinding(() -> graphicsOverlay.getGraphics().isEmpty())));
```
In Qt they check if the graphics overlay is empty so we should do that here too.